eth/db/header.py

```diff
@@ -46,6 +50,91 @@ class HeaderDB(HeaderDatabaseAPI):
     def __init__(self, db: AtomicDatabaseAPI) -> None:
         self.db = db
 
+    def get_chain_integrity_info(self) -> Tuple[Tuple[int, int], ...]:
+        return self._get_chain_integrity_info(self.db)
+
+    @classmethod
+    def _get_chain_integrity_info(cls, db: DatabaseAPI) -> Tuple[Tuple[int, int], ...]:
+        try:
+            encoded_integrity_info = db[SchemaV1.make_header_chain_integrity_info_lookup_key()]
+        except KeyError:
+            return ()
+        else:
+            return decode_chain_integrity_info(encoded_integrity_info)
+
+    @classmethod
+    def _update_chain_integrity_info(
+            cls,
+            db: DatabaseAPI,
+            persisted_header: BlockHeaderAPI,
+            base_integrity_info: Tuple[Tuple[int, int], ...] = None) -> Tuple[Tuple[int, int], ...]:
```

Comment:
You might alias `BlockRange = Tuple[BlockNumber, BlockNumber]`  (or `int` if it makes typing easier).